sim/priest/priest.go

```diff
@@ -18,11 +41,205 @@ func (priest *Priest) GetCharacter() *core.Character {
+func (priest *Priest) Init(sim *core.Simulation) {
+	priest.mindflayCastTemplate = priest.newMindflayTemplate(sim)
+	priest.mindblastCastTemplate = priest.newMindBlastTemplate(sim)
+	priest.swpCastTemplate = priest.newSWPTemplate(sim)
+	priest.vtCastTemplate = priest.newVTTemplate(sim)
+	priest.swdCastTemplate = priest.newSWDTemplate(sim)
+}
+
+func (priest *Priest) Reset(newsim *core.Simulation) {
+	// Cleanup and pending dots and casts
+	priest.mindblastSpell = core.SimpleSpell{}
+	priest.swdSpell = core.SimpleSpell{}
+	priest.MindFlaySpell = core.SimpleSpell{}
+	priest.SWPSpell = core.SimpleSpell{}
+	priest.VTSpell = core.SimpleSpell{}
+
+	priest.Character.Reset(newsim)
 }
 
-func (priest *Priest) Reset(sim *core.Simulation) {}
+func (priest *Priest) Advance(sim *core.Simulation, elapsedTime time.Duration) {
+	// druid should never be outside the 5s window, use combat regen.
+	priest.Character.RegenManaCasting(sim, elapsedTime)
+	priest.Character.Advance(sim, elapsedTime)
+}
 
 func (priest *Priest) Act(sim *core.Simulation) time.Duration {
-	return core.NeverExpires // makes the bot wait forever and do nothing.
+	return core.NeverExpires // does nothing
+}
+
+func (priest *Priest) applyTalentsToShadowSpell(cast *core.Cast, effect *core.SpellHitEffect) {
+	if cast.ActionID.SpellID == SpellIDMF || cast.ActionID.SpellID == SpellIDMB {
+		cast.ManaCost -= cast.BaseManaCost * float64(priest.Talents.FocusedMind) * 0.05
+	}
+	if cast.SpellSchool == stats.ShadowSpellPower {
+		effect.DamageMultiplier *= 1 + float64(priest.Talents.Darkness)*0.02
+
+		if priest.Talents.Shadowform {
+			effect.DamageMultiplier *= 1.15
+		}
+
+		// shadow focus gives 2% hit per level
+		effect.BonusSpellHitRating += float64(priest.Talents.ShadowFocus) * 2 * core.SpellHitRatingPerHitChance
+	}
+}
+
+func NewPriest(char core.Character, selfBuffs SelfBuffs, talents proto.PriestTalents) Priest {
+
+	// char.AddStat(stats.SpellHit, float64(talents.BalanceOfPower)*2*core.SpellHitRatingPerHitChance)
+
+	char.AddStatDependency(stats.StatDependency{
+		SourceStat:   stats.Intellect,
+		ModifiedStat: stats.SpellCrit,
+		Modifier: func(intellect float64, spellCrit float64) float64 {
+			return spellCrit + (intellect/79.4)*core.SpellCritRatingPerCritChance
+		},
+	})
+
+	if talents.Meditation > 0 {
+		char.PseudoStats.SpiritRegenRateCasting = float64(talents.Meditation) * 0.1
+	}
+
+	// if talents.LunarGuidance > 0 {
+	// 	bonus := (0.25 / 3) * float64(talents.LunarGuidance)
+	// 	char.AddStatDependency(stats.StatDependency{
+	// 		SourceStat:   stats.Intellect,
+	// 		ModifiedStat: stats.SpellPower,
+	// 		Modifier: func(intellect float64, spellPower float64) float64 {
+	// 			return spellPower + intellect*bonus
+	// 		},
+	// 	})
+	// }
+
+	priest := Priest{
+		Character: char,
+		SelfBuffs: selfBuffs,
+		Talents:   talents,
+	}
+
+	return priest
+}
+
+// newVTOnTick is the OnDamage function for all priest DoTs to apply VT
+func newVTOnTick(party *core.Party) core.OnDamageTick {
```

Comment:
This could be implemented entirely within vampiric_touch.go, as an aura that has OnSpellHit and OnDamageTick callbacks.

Reply:
we don't yet have OnDamageTick on auras. I think we should add that as part of our dot cleanup

Reply:
Adding it is like 5 lines of code   ;)